msbuild/Xamarin.MacDev.Tasks/Tasks/Sharpie.cs

```diff
@@ -0,0 +1,85 @@
+using System;
+using System.IO;
+using System.Runtime.InteropServices;
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Xamarin.MacDev.Tasks {
+	public class Sharpie : XamarinToolTaskWithOutput {
+		public override string TaskPrefix => "SHRP";
+
+		protected override string ToolName => "sharpie";
+
+		public string Command { get; set; } = string.Empty;
+
+		public string Arguments { get; set; } = string.Empty;
+
+		public string Namespace { get; set; } = string.Empty;
+
+		public string Sdk { get; set; } = string.Empty;
+
+		public string Scope { get; set; } = string.Empty;
+
+		public string OutputPath { get; set; } = string.Empty;
+
+		public ITaskItem [] Headers { get; set; } = Array.Empty<ITaskItem> ();
+
+
+		const string ClassicXIAssembly = "/Library/Frameworks/Xamarin.iOS.framework/Versions/Current/lib/64bits/iOS/Xamarin.iOS.dll";
+
+		protected override string GenerateFullPathToTool ()
+		{
+			return Path.Combine ("/usr", "local", "bin", ToolExe);
+		}
+
+		protected override string GenerateCommandLineCommands ()
+		{
+			var cmd = new CommandLineBuilder ();
+
+			if (!string.IsNullOrEmpty (Command))
+				cmd.AppendSwitch (Command);
+
+			if (!string.IsNullOrEmpty (Namespace))
+				cmd.AppendSwitchIfNotNull ("--namespace ", Namespace);
+
+			if (!string.IsNullOrEmpty (Sdk))
+				cmd.AppendSwitchIfNotNull ("--sdk ", Sdk);
+
+			if (!string.IsNullOrEmpty (Scope))
+				cmd.AppendSwitchIfNotNull ("--scope ", Scope);
+
+			if (!string.IsNullOrEmpty (OutputPath))
+				cmd.AppendSwitchIfNotNull ("--output ", OutputPath);
+
+			cmd.AppendSwitchIfNotNull (string.Empty, Headers, delimiter: " ");
+
+			if (!string.IsNullOrEmpty (Arguments))
+				cmd.AppendSwitch (Arguments);
+
+			return cmd.ToString ();
+		}
+
+		public override bool RunTask ()
+		{
+			if (RuntimeInformation.IsOSPlatform (OSPlatform.OSX)) {
+				if (!File.Exists (GenerateFullPathToTool ())) {
+					// TODO loc
+					Log.LogError ($"{TaskPrefix}1000 {{0}}", $"Unable to find the executable \"{ToolName}\". Please install Objective-Sharpie: https://aka.ms/objective-sharpie.");
+					return false;
+				}
+
+				if (!File.Exists (ClassicXIAssembly)) {
+					Log.LogError ($"{TaskPrefix}1001 {{0}}", $"The \"{ToolName}\" tool requires a classic Xamarin.iOS installation. Please install Xamarin.iOS: https://github.com/xamarin/xamarin-macios/blob/main/DOWNLOADS.md");
+					return false;
```

Comment:
My concern is the tool failing with the following, and that error case being less clear to the customer:

```
System.IO.FileNotFoundException: Invalid Image
File name: '/Library/Frameworks/Xamarin.iOS.framework/Versions/Current/lib/64bits/iOS/Xamarin.iOS.dll'
  at (wrapper managed-to-native) System.Reflection.Assembly.LoadFrom(string,bool,System.Threading.StackCrawlMark&)
  at System.Reflection.Assembly.LoadFrom (System.String assemblyFile) [0x00002] in /Users/builder/jenkins/workspace/build-package-osx-mono/2020-02/external/bockbuild/builds/mono-x64/mcs/class/corlib/System.Reflection/Assembly.cs:367 
  at Sharpie.Bind.Massagers.XamarinApiMappingMassager.Initialize () [0x0002c] in /Users/runner/work/1/s/Sharpie.Bind/Massagers/XamarinApiMappingMassager.cs:50 
```

When I was last looking like this I thought this was fatal, though in a couple simple tests cases I am running now this no longer seems to be a fatal error. 

Maybe it would be best to update this to a warning along the lines of "your mileage may vary at this time" and remove the download/install suggestion?

Reply:
I changed this to be a warning, can revisit if needed.

Reply:
This task has been removed